lib/matplotlib/axes/_base.py

```diff
@@ -180,8 +178,56 @@ def __call__(self, *args, **kwargs):
             if yunits != self.axes.yaxis.units:
                 self.axes.yaxis.set_units(yunits)
 
-        ret = self._grab_next_args(*args, **kwargs)
-        return ret
+        if not args:
+            return
+
+        data = kwargs.pop("data", None)
+        if data is not None:
+            replaced = [mpl._replacer(data, arg) for arg in args]
```

Comment:
This block is quite long. Can it be extracted into a private method?

Reply:
It's really not that long (half of it is error messages and comments explaning what's happening).  Also it's modifying args and kwargs in-place, which, sure, is also possible from a helper method, but I'm not convinced it helps with legibility.  (Literally, `__call__` does three things: 1) apply units, 2) handle the "data" kwarg, 3) grab two-or-three args at a time and pass them to the plotting functoin.  We may as well keep them in that order.)

Reply:
Ok, I agree that this can stay a single method. However, it definitively needs this explanation:

> Literally, __call__ does three things: 1) apply units, 2) handle the "data" kwarg, 3) grab two-or-three args at a time and pass them to the plotting functoin.

If that was in the code, it would already be much more understandable. Can you please add that to the docstring, or alternatively split it into comments before each of the sections.

Reply:
done